src/care/local_ptr.h

```diff
@@ -84,12 +84,16 @@ namespace care {
                    typename std::enable_if<B, int>::type = 1>
          CARE_HOST_DEVICE local_ptr<T>(host_ptr<T_non_const> const &ptr) noexcept : m_ptr(ptr) {}
 
+         ///
+         /// TODO: When CHAI is updated, change the following constructors to ptr.data() instead of ptr.getPointer()
+         /// 
+
          ///
          /// @author Peter Robinson
          ///
          /// Construct from chai::ManagedArray
          ///
-         CARE_HOST_DEVICE local_ptr<T>(chai::ManagedArray<T> const &ptr) : m_ptr(ptr) {}
+         CARE_HOST_DEVICE local_ptr<T>(chai::ManagedArray<T> const &ptr) : m_ptr(ptr.getPointer(chai::CPU)) {}
```

Comment:
Because this is a HOST_DEVICE function, you should use .data() with no parameters.

Reply:
Yes, that is true, but it will probably have to wait until CHAI is updated for that to happen. I will hold off on this for the time being